Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -886,6 +886,11 @@ open class NavigationMapView: UIView {
      */
     var mostRecentUserCourseViewLocation: CLLocation?
     
+    /**
+     The coordinates and corresponding identifiers for highlight buildings.
+     */
+    var highlightBuildings = [CLLocationCoordinate2D: Int64]()
```

Comment:
To store the coordinate and identifier pairs when the building is out of map boundary.

Reply:
Let’s give this a more descriptive name, such as `highlightedBuildingIdentifiersByCoordinate` to avoid sounding like a method like `highlightBuildings(at:in3D:extrudeAll:completion:)`.

A given building be found at multiple coordinates depending on the coordinates that happen to be passed into `highlightBuildings(at:in3D:extrudeAll:completion:)`. Is this potential duplication a problem?

Reply:
If a building could be found at multiple coordinates. If one of the coordinate is stored, and we try to highlight it with another coordinate. The previous one will be removed, and the new one will be queried and stored. And when we unhighlight buildings, all the coordinates stored will be removed. The property has key for coordinate, so there shouldn't be the duplication problem.